### PR TITLE
Schema renaming documentation

### PR DESCRIPTION
**Pull request by BobaFetters**
Adding documentation for the schema type renaming feature.

### CODE REVIEW

File: docs/source/code-generation/codegen-configuration.mdx [hunk 1]
```diff
@@ -445,7 +445,28 @@ The top-level properties are:
 		"inputObjects": "camelCase"
 	},
 	"pruneGeneratedFiles": true,
-	"markOperationDefinitionsAsFinal": true
+	"markOperationDefinitionsAsFinal": true,
+	"schemaCustomization" : {
+    "customTypeNames" : {
```
Comment (BobaFetters) [inline docs/source/code-generation/codegen-configuration.mdx:450]: Not sure what's going on with the indentation here and below, double checked it locally and it looks fine.

**Review by AnthonyMDev — APPROVED** [review]
lgtm

File: docs/source/code-generation/codegen-configuration.mdx [hunk 2]
```diff
@@ -475,7 +496,105 @@ let configuration = ApolloCodegenConfiguration(
...
+)
+```
+
+</MultiCodeBlock>
+
+### Schema Customization
+
+#### Custom Type Names
+
+Schema types can have their names used in Swift code customized by using the [`customTypeNames`](https://www.apollographql.com/docs/ios/docc/documentation/apollocodegenlib/apollocodegenconfiguration/schemacustomization/customtypenames) option under [`schemaCustomization`](https://www.apollographql.com/docs/ios/docc/documentation/apollocodegenlib/apollocodegenconfiguration/schemacustomization) to provide new names for the types which you wish to rename in Swift. This does not affect your schema directly, or network requests sent to your server, only the type names used by you in your Swift code. The `customTypeNames` option accepts a Dictionary which maps schema type names given as a `String` to a [`CustomSchemaTypeName`](https://www.apollographql.com/docs/ios/docc/documentation/apollocodegenlib/apollocodegenconfiguration/schemacustomization/customschematypename) enum case with associated values for how to customize the given schema type.
```
Comment (calvincestari) [inline docs/source/code-generation/codegen-configuration.mdx:529]:
```suggestion
Schema types can have their names used in Swift code customized by using the [`customTypeNames`](https://www.apollographql.com/docs/ios/docc/documentation/apollocodegenlib/apollocodegenconfiguration/schemacustomization/customtypenames) option under [`schemaCustomization`](https://www.apollographql.com/docs/ios/docc/documentation/apollocodegenlib/apollocodegenconfiguration/schemacustomization) to provide new names for the types which you wish to rename in Swift. This does not affect your schema directly, or network requests sent to your server, only the type names used by you in your Swift code. The `customTypeNames` option accepts a `Dictionary` which maps schema type names given as a `String` to a [`CustomSchemaTypeName`](https://www.apollographql.com/docs/ios/docc/documentation/apollocodegenlib/apollocodegenconfiguration/schemacustomization/customschematypename) enum case with associated values for how to customize the given schema type.
```

File: docs/source/code-generation/codegen-configuration.mdx [hunk 3]
```diff
@@ -475,7 +496,105 @@ let configuration = ApolloCodegenConfiguration(
...
+- Object
+- Union
+
+The `CustomSchemaTypeName` enum contains the following possible cases:
+
+| Case | Descriptions |
+| ---- | ------------ |
+| `.type(name:)` | This case can be used to customize the name of any type in your schema by supplying a `String` to the `name` parameter representing the desired new name. |
+| `.enum(name:cases:)` | This case must be used when you want to customize the cases of an enum type. You can optionally supply a `String` to the `name` parameter to customize the name of the enum itself, and supply a `[String: String]` dictionary to the `cases` parameter to customize any cases you desire. |
+| `.inputObject(name:fields:)` | This case must be used when you want to customize the fields of an input object type. You can optionally supply a `String` to the `name` parameter to customize the name of the input object itself, and supply a `[String: String]` dictionary to the `fields` parameter to customize any fields you desire. |
```
Comment (calvincestari) [inline docs/source/code-generation/codegen-configuration.mdx:545]:
```suggestion
| `.inputObject(name:fields:)` | This case must be used when you want to customize the fields of an Input Object type. You can optionally supply a `String` to the `name` parameter to customize the name of the Input Object itself, and supply a `[String: String]` dictionary to the `fields` parameter to customize any fields you desire. |
```

**Review by calvincestari — APPROVED** [review]
Minor text styling comments - LGTM!

**Pull request merged**